homeassistant/components/scene/__init__.py

```diff
@@ -0,0 +1,76 @@
+"""
+homeassistant.components.scene
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Allows users to set and activate scenes.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/scene/
+"""
+import logging
+from collections import namedtuple
+
+from homeassistant.const import (
+    ATTR_ENTITY_ID, SERVICE_TURN_ON, CONF_PLATFORM)
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.entity_component import EntityComponent
+
+DOMAIN = 'scene'
+DEPENDENCIES = ['group']
+STATE = 'scening'
+
+CONF_ENTITIES = "entities"
+
+SceneConfig = namedtuple('SceneConfig', ['name', 'states'])
+
+
+def activate(hass, entity_id=None):
+    """ Activate a scene. """
+    data = {}
+
+    if entity_id:
+        data[ATTR_ENTITY_ID] = entity_id
+
+    hass.services.call(DOMAIN, SERVICE_TURN_ON, data)
+
+
+def setup(hass, config):
+    """ Sets up scenes. """
+
+    logger = logging.getLogger(__name__)
+
+    for entry in config:
+        if DOMAIN in entry:
+            if not any(CONF_PLATFORM in key for key in config[entry]):
+                config[entry] = {'platform': 'homeassistant',
```

Comment:
Did you test this code locally? I am pretty sure that this will fail because the config that is passed in when loading HA normally is immutable.